src/xtopcom/xvledger/src/xvblock.cpp

```diff
@@ -2351,15 +2352,17 @@ namespace top
                 xassert(qcert_ptr != NULL); //should has value
                 if(qcert_ptr != NULL)
                 {
-                    const std::string vcert_hash = qcert_ptr->hash(vqcert_bin);
-                    if( (m_cert_hash.empty() == false) && (vcert_hash != m_cert_hash) ) //test match
-                    {
-                        xerror("xvblock_t::do_read, vqcert not match with stored hash, vcert_hash:%s but ask %s",vcert_hash.c_str(),m_cert_hash.c_str());
-                    }
-                    else
-                    {
-                        m_vqcert_ptr = qcert_ptr;
-                    }
+                    #if DEBUG
```

Comment:
do not check cert hash in release version?

Reply:
it's must check cert hash many times before do read.